src/pkg/clouds/aws/alb.go

```diff
@@ -0,0 +1,47 @@
+package aws
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	elbv2 "github.com/aws/aws-sdk-go-v2/service/elasticloadbalancingv2"
+)
+
+var AlbOverride AlbAPI
+
+type AlbAPI interface {
+	DescribeLoadBalancers(ctx context.Context, params *elbv2.DescribeLoadBalancersInput, optFns ...func(*elbv2.Options)) (*elbv2.DescribeLoadBalancersOutput, error)
+}
+
+func newAlbSvc(cfg aws.Config) AlbAPI {
+	var svc AlbAPI = AlbOverride
+
+	if svc == nil {
+		svc = elbv2.NewFromConfig(cfg)
+	}
+
+	return svc
+}
+
+func (a *Aws) GetAlbDnsName(ctx context.Context, name string) (string, error) {
```

Comment:
not being called?

Reply:
Yes, they are not used anymore, it is left here as funtions potentially useful in the future. I can remove them too.
The old design is to [reconstruct the alb name](https://github.com/DefangLabs/defang/pull/915/commits/6ae8a6f084ea81a5f0a79e93106269916774eed5) and use this function to get its DNS name, but the updated design is to update the protobuf, and let CD to set the lb_dns_name like the existing endpoints [in byoc.ts](https://github.com/DefangLabs/defang-mvp/blob/main/pulumi/cd/aws/byoc.ts#L59-L66).